lua/package-info/ui/generic/loading-status.lua

```diff
@@ -121,41 +126,44 @@ M.update_spinner = function(message)
         M.state.notification = new_notif
     end
 
-    vim.fn.timer_start(60, function()
-        M.update_spinner()
+    -- this can be used to post updates (ex. refresh the statusline)
+    vim.schedule(function()
+        vim.api.nvim_exec_autocmds('User', {
```

Comment:
What does `User` do in this case?

Reply:
User is a [special event that lets execute user defined events](https://neovim.io/doc/user/autocmd.html#_5.-events), since Nim doesn't let you register your own custom events (or I didn't find a way to do so). If there's a better way, lmk for sure